### PR TITLE
Prevent plugins from being loaded multiple times

### PR DESCRIPTION
**Pull request by dcrissman**

### CODE REVIEW

File: common/src/main/java/com/redhat/lightblue/rest/RestConfiguration.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ public static LightblueFactory getFactory() {
         return factory;
     }
 
-    public static LightblueFactory getFactory(
+    public synchronized static LightblueFactory getFactory(
```
Comment (bserdar) [inline common/src/main/java/com/redhat/lightblue/rest/RestConfiguration.java]: Inefficient. Instead:
1) Declare factory as volatile
2) remove synchronized
3) getFactory() {
       LightblueFactory f=factory;
       if(f==null) {
            synchronized(this) {
                    if(factory==null)  initialize factory
                    f=factory;
            }
   return f;
}

Comment (dcrissman) [inline common/src/main/java/com/redhat/lightblue/rest/RestConfiguration.java]: > synchronized(this)

That was my initial thought also, but we are in a static context so it won't work. I could create a lock object also in a volatile state, but I guess that seemed silly over just making the method synchronized.

File: common/src/main/java/com/redhat/lightblue/rest/RestConfiguration.java [hunk 2]
```diff
@@ -60,36 +60,40 @@ public static DataSourcesConfiguration getDatasources() {
...
+        LightblueFactory f = factory;
+        if (f == null) {
             return getFactory(loadDefaultPlugins());
         }
-        return factory;
+        return f;
     }
 
-    public static LightblueFactory getFactory(
+    public synchronized static LightblueFactory getFactory(
```
Comment (dcrissman) [inline common/src/main/java/com/redhat/lightblue/rest/RestConfiguration.java]: Still an open discussion about synchronized method vs. using a synchronized block.

File: common/src/main/java/com/redhat/lightblue/rest/RestConfiguration.java (hunk 1 above)
Comment (bserdar) [inline common/src/main/java/com/redhat/lightblue/rest/RestConfiguration.java]: You can use synchronize on RestConfiguration.class.

**Comment by dcrissman** [comment]
@bserdar ready for review.

**Pull request merged**